logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/incremental/EventGatherer.kt

```diff
@@ -96,22 +96,17 @@ internal class EventGathererImpl(
     private suspend fun FlowCollector<Event>.handleWebsocketEvent(webSocketEvent: WebSocketEvent<Event>) = when (webSocketEvent) {
         is WebSocketEvent.Open -> onWebSocketOpen()
         is WebSocketEvent.BinaryPayloadReceived -> onWebSocketEventReceived(webSocketEvent)
-        is WebSocketEvent.Close -> throwOnWebSocketClosed(webSocketEvent)
+        is WebSocketEvent.Close -> handleWebSocketClosure(webSocketEvent)
         is WebSocketEvent.NonBinaryPayloadReceived -> logger.w("Non binary event received on Websocket")
     }
 
-    private fun throwOnWebSocketClosed(webSocketEvent: WebSocketEvent.Close<Event>): Nothing =
-        throw when (val cause = webSocketEvent.cause) {
+    private fun handleWebSocketClosure(webSocketEvent: WebSocketEvent.Close<Event>) =
+        when (val cause = webSocketEvent.cause) {
+            null -> logger.i("Websocket closed normally")
```

Comment:
Is that really a normal condition? I mean isn't better to mention that the cause is null for further investigation?

Reply:
This means that either us chose to close it, by calling cancel, or the backend close to do so by sending a close websocket message.

AFAIK the server doesn't do it. And we only do it when the connection policy is `DISCONNECT_AFTER_PENDING_EVENTS`.

So indeed, it's a normal/expected case.